recipe/build.sh

```diff
@@ -1,26 +1,64 @@
+#!/bin/bash
+set -ex
+
 # Removes -std=c++XX from build ${CXXFLAGS}.
 # Ignores irrelevant Clang availability annotations on MacOS: 
 # https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
 export CXXFLAGS="$(echo "${CXXFLAGS}" | sed -E 's@-std=c\+\+[^ ]+@@g') -D_LIBCPP_DISABLE_AVAILABILITY"
 
-# Get an updated config.sub and config.guess
-cp $BUILD_PREFIX/share/gnuconfig/config.* .
+if [[ "$SHLIB_EXT" == '.dylib' ]]; then
+
+  if [[ "${CONDA_BUILD_CROSS_COMPILATION}" != "1" ]]; then
+  BUILD_TEST=ON
+  else
+  BUILD_TEST=OFF
+  fi
+
+  export CMAKE_BUILD_TYPE=Release
+  export BUILD_SHARED_LIBS=ON
+  export CMAKE_BUILD_WITH_INSTALL_RPATH=ON
+  export CMAKE_CXX_STANDARD=17
+  export TMPDIR="${SRC_DIR}\build\tmp"
+
+  mkdir build
+  pushd build
+  mkdir tmp
+
+  cmake -GNinja ${CMAKE_ARGS} \
+      -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+      -DCMAKE_PREFIX_PATH="${PREFIX}" \
+      -DOPENFST_BUILD_TEST=$BUILD_TEST \
+      ..
+
 
-if [[ "${CONDA_BUILD_CROSS_COMPILATION}" == "1" ]]; then
-  OPENFST_CROSS_COMPILATION_CONFIGURE_OPTS="--build=${BUILD} --host=${HOST}"
+  cmake --build . --verbose --config Release -- -v -j ${CPU_COUNT}
+  ctest
+  cmake --install . --verbose --config Release
+
+
+  popd
 else
-  OPENFST_CROSS_COMPILATION_CONFIGURE_OPTS=""
-fi
+  # Get an updated config.sub and config.guess
+  cp $BUILD_PREFIX/share/gnuconfig/config.* .
+
+  ./configure \
+     --prefix="${PREFIX}" \
+     --enable-static=no \
+     --enable-compress \
+     --enable-compact-fsts \
+     --enable-fsts \
+     --enable-far \
+     --enable-grm \
```

Comment:
Could you restore the old list of `--enable-foo`s? The current one is redundant, as you can confirm if you read the code, since `--enable-grm` is an alias for all the ones you added.

Reply:
There's some difference in behavior for installing header files between the --enable-fsts and explicitly calling --enable-*-fsts.  With just --enable-fsts, it doesn't install include/fst/const, include/fst/compact, include/fst/linear, include/fst/lookahead, or include/fst/ngram, which I added tests for in the working windows build, so it's likely not a new issue, just undetected.

Reply:
> There's some difference in behavior for installing header files between the `--enable-fsts` and explicitly calling `--enable-*-fsts`. With just `--enable-fsts`, it doesn't install `include/fst/const`, `include/fst/compact`, `include/fst/linear`, `include/fst/lookahead`, or `include/fst/ngram`, which I added tests for in the working windows build, so it's likely not a new issue, just undetected.

It doesn't necessarily follow that we should change the behavior here. There are *a lot* of unrelated changes and I'd prefer them in a separate PR. Is it not general practice to isolate changes? 

Anyways, the thing about --enable-grm remains true. I looked at `src/include/Makefile.am` and that definitely installs the FAR headers if you --enable-grm.

Reply:
Found the cause of the missing headers, so I'll make a quick patch for those.